BitmovinConvivaAnalytics/Classes/ConvivaAnalytics.swift

```diff
@@ -385,26 +385,62 @@ public final class ConvivaAnalytics: NSObject {
             value: Int64(player.currentTime(.relativeTime) * 1_000)
         )
     }
-}
 
-private extension Ad {
-    var adInfo: [String: Any] {
+    private func buildAdInfo(adStartedEvent: AdStartedEvent) -> [String: Any] {
         var adInfo = [String: Any]()
+
+        adInfo["c3.ad.id"] = "NA"
+        adInfo["c3.ad.system"] = "NA"
+        adInfo["c3.ad.mediaFileApiFramework"] = "NA"
+        adInfo["c3.ad.firstAdSystem"] = "NA"
+        adInfo["c3.ad.firstAdId"] = "NA"
+        adInfo["c3.ad.firstCreativeId"] = "NA"
+
         adInfo["c3.ad.technology"] = "Client Side"
+        if adStartedEvent.clientType == AdSourceType.ima {
+            adInfo[CIS_SSDK_PLAYER_FRAMEWORK_NAME] = "Google IMA SDK"
+            adInfo[CIS_SSDK_PLAYER_FRAMEWORK_VERSION] = contentMetadataBuilder.metadataOverrides.imaSdkVerison ?? "NA"
+        } else {
+            adInfo[CIS_SSDK_PLAYER_FRAMEWORK_NAME] = "Bitmovin"
+            adInfo[CIS_SSDK_PLAYER_FRAMEWORK_VERSION] = playerHelper.version
+        }
 
-        if mediaFileUrl != nil {
-            adInfo[CIS_SSDK_METADATA_STREAM_URL] = mediaFileUrl
+        adInfo["c3.ad.position"] = AdEventUtil.parseAdPosition(
+            event: adStartedEvent,
+            contentDuration: player.duration
+        ).rawValue
+        adInfo[CIS_SSDK_METADATA_DURATION] = adStartedEvent.duration
+        adInfo[CIS_SSDK_METADATA_IS_LIVE] = videoAnalytics.getMetadataInfo()[CIS_SSDK_METADATA_IS_LIVE]
+
+        let ad = adStartedEvent.ad
```

Comment:
Let's exclude the term `ad` from being too short in SwiftLint.

Reply:
Sorry for cramping the linting changes in this PR

https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/71/commits/6f66f9cb87585dd6475a9f51d19efc1e7218c4db
https://github.com/bitmovin/bitmovin-player-ios-analytics-conviva/pull/71/commits/e792261c06395da323b11f193184a717bea8c3dd